test/npu-xrt/add_one_ctrl_packet/test.cpp

```diff
@@ -145,7 +145,7 @@ int main(int argc, const char *argv[]) {
 
   uint32_t beats = 1 - 1;
   uint32_t operation = 0;
-  uint32_t stream_id = 0;
+  uint32_t stream_id = 5;
```

Comment:
I didn't think this had any effect for writes without return packets, i.e. the first two packets in this test. Did you find otherwise?

Reply:
Yeah with this PR the `stream_id` must match the controller id of tile(0, 2), because the pass `aie-generate-column-control-overlay` also generates the packet-flow routing between tile(0, 2) CTRL and tile(0, 0) SOUTH0 for TCT routing, with packet header being controller id. 

I got deadlock if I do not change the stream_id to match with controller id of tile(0, 2).

Reply:
Maybe having the `aie-generate-column-control-overlay` generating TCT routings for all tiles to shim SOUTH0 is dangerous... Let me see if I limit this pass to route TCT from SHIM CTRL only...

Reply:
> I'm referring to the shim -> tile control packets, which are the ones getting stream_id = 5 here. For these the packet header id is set by the shim bd in the block write. For the return packets of the control port read, the controller id gets set to 2 in the code below (line 181), and that is the id which must match the routing configuration.

I see. You are right. After I limit the pass to generate TCT routings for shim tiles only, then stream_id again resumed to having no effect.